### DISCUSSION TITLE
Customize json encoding by field name?

### DISCUSSION THREAD

**Discussion by kmorris81** [pydantic/pydantic#2596] [Question] [answered]
```
my_float2: Optional[float] = Field(None,description='My float2',)
my_float5: Optional[float] = Field(None,description='My float5',)
```

When I call  `MyFloatClass.json()` I would like for the value of `my_float1` to be rounded to a precision of two decimal places and the value of `my_float2` to five decimal places.

I see how to override types using the `class Config` `json_encoder`, however, since both fields use the same type I need to differentiate them by name.  Is this possible?

Thanks!

**Comment by kmorris81** [comment, marked as answer]
Naturally, I answered my own question shortly after asking.  
The answer is to use `@validators`

**Reply by jenia** [reply]
But what if you want to *calculate* the field value like this: https://github.com/samuelcolvin/pydantic/issues/2627
